contracts/src/lib/FROST.sol

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.30;
+
+import {Secp256k1} from "@/lib/Secp256k1.sol";
+
+/// @title FROST
+/// @notice Implementation of FROST(secp256k1, SHA-256) cyphersuite.
+library FROST {
+    using Secp256k1 for Secp256k1.Point;
+
+    struct Commitment {
+        uint256 index;
+        Secp256k1.Point d;
+        Secp256k1.Point e;
+    }
+
+    error UnorderedCommitments();
+    error InvalidScalar();
+
+    /// @notice Generate a random nonce from some randomness and a secret key.
+    function nonce(bytes32 random, Secp256k1.Point memory secret) internal view returns (uint256 n) {
```

Comment:
Are you sure that this method is correctly implemented? 

As far as I understand the `nonce_generate(secret)` function definition, `secret` is not the public key (so a point), but rather the secret key (so a uint256). 

As far as I understand you signing test (https://github.com/safe-research/shieldnet/pull/16/files#diff-99c3b88eb27150979236c3a222c847c6dd5f6e444dd739b42183db3ecfb52362R236) you use g(s_1), which would be the public key, right?

Reply:
https://datatracker.ietf.org/doc/html/rfc9591#section-6.5 links to https://secg.org/sec1-v2.pdf. On page 12 we see that this points for secp256k1 (as it p is odd, so case 1) to use  Integer-to-Octet-String Conversion, which is